parser/parser.go

```diff
@@ -273,14 +275,25 @@ func (p *Parser) expectPeek(t token.Type) bool {
 	return false
 }
 
+// ErrorContext returns the current line and a pointer to the error position.
+// If prev is true, the error position is relative to the previous token instead of current one.
+func (p *Parser) ErrorContext(prev bool) string {
```

Comment:
Two remarks:
- context is connoted in Go. I would have avoided
- it's not really about error. You do use it to say where the error is, but there is nothing about error here. The function can be called even when there is no error

Reply:
Maybe tgis method could call Errors() (it's the right object, right ?) and return a different result when there is no error

Reply:
it's used to construct the errors so it can't use the errors